### ISSUE TITLE
Test failures in scipy.special with numpy master

### ISSUE THREAD

**Issue by WarrenWeckesser** [scipy/scipy#7907]
In recent pull requests, a couple tests in scipy.special are failing with numpy master.  One of the tests is effectively this:

```
>>> from scipy.special import iv
>>> v, z = (-120, 700.6)
>>> assert_allclose(iv(v, z), iv(v, z+0j), rtol=1e-11, atol=1e-305)
```

In scipy `'1.1.0.dev0+04759f9'`, that assertion passes with numpy 1.13.1, but raises an `AssertionError` with numpy `'1.14.0.dev0+54232da'`.

Apparently there is a change in the behavior of `np.allclose()`.  I have reported the issue here: https://github.com/numpy/numpy/issues/9734

**Comment by pv** [comment]
Fixed in numpy/numpy#9741

**Comment by WarrenWeckesser** [comment]
Thanks Pauli.
